librecad/src/lib/engine/rs_ellipse.cpp

```diff
@@ -799,6 +799,58 @@ bool RS_Ellipse::createFromQuadratic(const QVector<double>& dn){
 }
 
 /**
+ * The general equation of a conic is Ax^2 + Bxy + Cy^2 + Dx + Ey + F = 0
+ * In matrix form it is: {x, y}.m_mQuad.{{x},{y}} + m_vLinear.{{x},{y}}+m_dConst=0
+ * If the det(m_mQuad) < 0,  then the conic is an ellipse.
```

Comment:
consider any 2D vector in the xy-plane, {x, y}, our quadratic has the form of: { x, y}.{ { A, C/2}, {C/2, B}}.{ {x}, {y}} = Ax^2+Bxy+Cy^2. The requirement of the original equaiton being an ellipse means the quadratic matrix is positive definite, i.e., Ax^2 +Cxy + By^2 >0, for any {x, y}, it means: A>0, C>0, and C^2 - 4AB<0 ( det( { {A, C/2}, {C/2, B}}) >0